[7.0] Run robotest with regular user permissions.

wadells opened the pull request:
## Description
This is a quick backport of the CI permissions fix to #1915, introduced in #2133.

I'm front-running because that port is large and won't necessarily go to all branches (still need to discuss this), and this one-liner provides excellent value in the interim.

## Type of change
* Internal change (not necessarily a bug fix or a new feature)

## Linked tickets and other PRs
Contributes to #1915.

## TODOs
- [x] Self-review the change
- [x] Verify the content of this PR build directory is 100% owned by Jenkins
- [x] Address review feedback

## Testing done
None, the pr build is sufficient.

wadells committed:
Run robotest with regular user permissions.

Contributes to #1915.

a-palchikov reviewed (approved):
Might be worth to look into running docker [rootless](https://docs.docker.com/engine/security/rootless/#rootless-docker-in-docker)

wadells commented:
> Might be worth to look into running docker [rootless](https://docs.docker.com/engine/security/rootless/#rootless-docker-in-docker)

Thanks! I'll take a look at this as I backport the more substantial changes in #2133.

wadells committed:
Preempt docker creation of bin dir owned by root:root.

This is a total hack to prevent docker from creating a mountpoint owned
by root (which then can't be deleted by Jenkins). The long term fix will
need to be by fixing or no longer relying on robotest's run_suite.sh.

wadells commented:
Jenkins build file ownership is clean:

```
[jenkins@jenkins Gravity-PR-5.2_PR-2152-RWZWLQ4PHPWEYTC3LTZRWFRAEIWM2ECDND7M6C27NGN3HTFWVVQQ]$ find . \! -user $(id -u) -print 
[jenkins@jenkins Gravity-PR-5.2_PR-2152-RWZWLQ4PHPWEYTC3LTZRWFRAEIWM2ECDND7M6C27NGN3HTFWVVQQ]$ 
````
